test/tools/ossfuzz/protoToYul.h

```diff
@@ -295,12 +309,16 @@ class ProtoConverter
 	}
 
 	std::ostringstream m_output;
-	/// Variables in current scope
-	std::stack<std::vector<std::string>> m_scopeVars;
+	/// Variables in all function definitions
+	std::vector<std::vector<std::vector<std::string>>> m_funcVars;
+	/// Variables in current function definition
+	std::vector<std::string const*> m_currentFuncVars;
+	/// Variables in global scope
+	std::vector<std::string const*> m_currentGlobalVars;
 	/// Functions in current scope
-	std::stack<std::vector<std::string>> m_scopeFuncs;
+	std::vector<std::vector<std::string>> m_scopeFuncs;
 	/// Variables
-	std::vector<std::string> m_variables;
+	std::vector<std::vector<std::string>> m_globalVars;
```

Comment:
Why a 2d vector instead of 1?

Reply:
Each global scope creates a 1d vector. So we have a vector of those scopes here

Example:

```
{ let x { let y { let z, a }}}
```

creates the following 2d vector {{x}, {y}, {z, a}} and depending on where we are in the "stack" we obtain variables that are live. For example, in the inner most scope, all variables are in scope post the `let z, a` statement.

Reply:
Ok!